### PR TITLE
Add Remarkable Desktop app 2.8.2

### PR DESCRIPTION
**Pull request by gianpaj**
- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

**new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/remarkable.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+cask "remarkable" do
+  version "2.8.2.170"
+  sha256 "7669f1e8dd1b13fd5305af8a215b17b33b7ff6cc218a218c9992b7dd83223465"
+
+  url "https://downloads.remarkable.com/desktop/production/mac/reMarkable-#{version}.dmg"
```
Comment (miccal) [inline Casks/remarkable.rb:5]: How does one find this `url` from the `homepage`?

**Comment by bevanjkay** [comment]
There is an appcast URL available at `https://get-updates.cloud.remarkable.engineering/sparkle/reMarkableMacOs/Prod/appcast.xml` - However the download URL provided by sparkle is different and it is hard to ascertain which parts would be dynamic until there is a new update available.

@gianpaj The cask should also include `auto_updates true`

It also seems that generally the app download is walled, however the download URL is active without any authorisation.

See: https://support.remarkable.com/hc/en-us/articles/360002665378-Desktop-app

**Comment by miccal** [comment]
Also see previous PR's: https://github.com/Homebrew/homebrew-cask/pull/102776 https://github.com/Homebrew/homebrew-cask/pull/45534 https://github.com/Homebrew/homebrew-cask/pull/45531 https://github.com/Homebrew/homebrew-cask/pull/39286

**Comment by vitorgalvao** [comment]
> It also seems that generally the app download is walled

Note that to be considered a walled download, it needs to be behind a login wall *and* be hosted on a different domain than the homepage. The second part doesn’t happen here, thus it’s not walled.

**Comment by miccal** [comment]
Thank you for your contribution to `homebrew-cask` @gianpaj.

Could you please try to add a [`zap` stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap), thanks.

**Comment by gianpaj** [comment]
@miccal how do I find which files should be zap'ed? Is there a way `brew` can automatically generate/find these files for me to add them to the cask definition?
And if so is there an example? I don't seem to understand the doc you provided. Feel free to update my PR if you're able to

**Comment by miccal** [comment]
> @miccal how do I find which files should be zap'ed? Is there a way `brew` can automatically generate/find these files for me to add them to the cask definition?
> And if so is there an example? I don't seem to understand the doc you provided. Feel free to update my PR if you're able to

The documentation I linked explains very clearly what to do, and includes an example for the Dropbox Cask.

**Comment by gianpaj** [comment]
I see now. That documentation text is hard to read with tired eyes... 

I've added the trash files now @miccal 

Thanks

**Comment by bevanjkay** [comment]
Thank you @gianpaj 
In future, please run `brew style --fix <cask>` before committing.

**Comment by gianpaj** [comment]
Got it. Thanks, guys!

BTW a little Github action checking the output of `brew style --fix <cask>` would be nice to have 🙃

**Comment by adavidzh** [comment]
I was reading this thread to understand how to go about getting version `2.10` in. IIUC:
- Regarding [this comment](https://github.com/Homebrew/homebrew-cask/pull/111097#issuecomment-916302748) on the question of being walled, be as it may that the link is not readily available, it is a fact that [version `2.10`](https://downloads.remarkable.com/desktop/production/mac/reMarkable-2.10.0.172.dmg) can be downloaded by anyone (with `wget`) and is hosted under `remarkable.com`.
- Regarding [this comment](https://github.com/Homebrew/homebrew-cask/pull/111097#issuecomment-915690182) on `auto_updates true`: AFAICT, none of these application versions are self-updatable. At least `2.5` wasn't and `2.10` isn't either.
- I checked https://get-updates.cloud.remarkable.engineering/sparkle/reMarkableMacOs/Prod/appcast.xml and it proposes both `2.8.2.170` and `2.10.0.172 ` (how to choose?) so I was wondering if that is a better path somehow; I've never worked with `appcast` in formulae.

With these in mind, what do you all think would be the best way to transmogrify (and update) this cask?

**Comment by bevanjkay** [comment]
> I checked https://get-updates.cloud.remarkable.engineering/sparkle/reMarkableMacOs/Prod/appcast.xml and it proposes both 2.8.2.170 and 2.10.0.172 (how to choose?) so I was wondering if that is a better path somehow; I've never worked with appcast in formulae.

@adavidzh If you look closely at the URL provided in the appcast for version `2.10.0.172` it is in the `dev` directory. This would indicate that it isn't the current stable release. However the app does seem to be returning this as the latest version with the in-app updater.

> Regarding this comment on auto_updates true: AFAICT, none of these application versions are self-updatable. At least 2.5 wasn't and 2.10 isn't either.

Version `2.8.2.170` definitely does allow checking for updates inside the app.

**Comment by adavidzh** [comment]
Thanks @bevanjkay:

> @adavidzh If you look closely at the URL provided in the appcast for version `2.10.0.172` it is in the `dev` directory. This would indicate that it isn't the current stable release. However the app does seem to be returning this as the latest version with the in-app updater.

That is quite odd in the sense that I was just proposed to download that from `https://downloads.remarkable.com/desktop/production/mac/reMarkable-2.10.0.172.dmg` where there is no `dev` hint. I wonder if the company proposes different versions depending on what they know is installed in the device...

> Version `2.8.2.170` definitely does allow checking for updates inside the app.

Perhaps I am not looking in the right place; where do you see that in the app? In `2.10.0.172` I see nothing on the menus and the `About reMarkable` window also doesn't show something like that.

**Pull request merged**